### PR TITLE
[component] add LeveledMeterProvider

### PR DESCRIPTION
**Pull request by codeboten**
This func will MeterProvider and MetricsLevel in the near future. Split from https://github.com/open-telemetry/opentelemetry-collector/pull/10912

### CODE REVIEW

File: .chloggen/codeboten_leveled-meter-split-2.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+change_type:
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: component
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add `TelemetrySettings.LeveledMeterProvider` func to replace MetricsLevel in the near future"
+
+# One or more tracking issues or pull requests related to the change
+issues: []
```
Comment (codeboten) [inline .chloggen/codeboten_leveled-meter-split-2.yaml:13]:
```suggestion
issues: [10931]
```

File: .chloggen/codeboten_leveled-meter-split-2.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type:
```
Comment (codeboten) [inline .chloggen/codeboten_leveled-meter-split-2.yaml:4]:
```suggestion
change_type: enhancement
```

File: service/service.go [hunk 3]
```diff
@@ -127,6 +128,12 @@ func New(ctx context.Context, set Settings, cfg Config) (*Service, error) {
 
 	logsAboutMeterProvider(logger, cfg.Telemetry.Metrics, mp, extendedConfig)
 	srv.telemetrySettings = component.TelemetrySettings{
+		LeveledMeterProvider: func(level configtelemetry.Level) metric.MeterProvider {
```
Comment (TylerHelmuth) [inline service/service.go:131]: Will the tests for these come later?

**Review by TylerHelmuth — APPROVED** [review]

File: service/service.go (hunk 3 above)
Comment (codeboten) [reply service/service.go:131]: yeah, when i move the code that uses the MeterProvider today to use the LeveledMeterProvider instead

**Pull request merged**